src/index.js

```diff
@@ -1064,6 +1287,21 @@ class Offline {
         resolve(this.server);
       });
     });
+    await new Promise((resolve, reject) => {
+      this.wsServer.start(err => {
```

Comment:
do we need to spin up another `hapi` instance? the examples seem to use the same: https://github.com/rse/hapi-plugin-websocket

Reply:
About `context` and `event` objects, I just did the minimum needed for the tests to pass. In order to have full support, I'll need to learn first how does serverless-offline handles these objects in the http side of things.

Reply:
makes sense.

Reply:
as for the 2 hapi instances: I believe we can just use port-sharing with 1 instance (the websocket plugin seems to be able to do so).

Reply:
The reason to spin up another `hapi` instance on port+1 is to mimic AWS in which it spins a new endpoint for WebSocket that runs on a different URL than for HTTP requests. Consolidating 2 endpoints into one is most probably error prone, e.g., how to handle when having both WS and HTTP routes on the root path. In this case, will localhost:3000/ be ws:// or http://?

Reply:
good point. Although, if collision could be a problem, we could perhaps add an additional root path for the sockets. (or randomly generated). e.g. ws://localhost:3000/__some-random-string-or-hardcoded__/my-route vs. http://localhost:3000/my-route . just a thought - might be not feasible.

Reply:
just checked the socket plugin docs. under: /*  provide combined REST/WebSocket route  */ seems to be able to handle cases like that as well. not sure about the feasibility for the `serverless-offline` plugin tho. would need to look a little deeper.

Reply:
event and context injection are now supported.